nixos/tests/blocky.nix

```diff
@@ -0,0 +1,54 @@
+import ./make-test-python.nix {
+  name = "blocky";
+
+  nodes = {
+    minimalConf = { ... }: {
+      services.blocky = {
+        enable = true;
+
+        settings = {
+          upstream = {
+            default = [ "8.8.8.8" "1.1.1.1" ];
+          };
+          port = 53;
+          httpPort = 4000;
+          logLevel = "info";
+        };
+      };
+    };
+
+    biggerConf = { pkgs, ... }: {
+      environment.systemPackages = [ pkgs.dnsutils ];
+      services.blocky = {
+        enable = true;
+
+        settings = {
+          customDNS = {
+            mapping = {
+              "printer.lan" = "192.168.178.3,2001:0db8:85a3:08d3:1319:8a2e:0370:7344";
+            };
+          };
+          upstream = {
+            default = [ "8.8.8.8" "1.1.1.1" ];
+          };
+          port = 53;
+          httpPort = 5000;
+          logLevel = "info";
+        };
+      };
+    };
+  };
+
+  testScript = ''
+    with subtest("Minimal config test"):
+        minimalConf.wait_for_unit("blocky.service")
+        minimalConf.wait_for_open_port(53)
+        minimalConf.wait_for_open_port(4000)
+
+    with subtest("Bigger config test"):
+        biggerConf.wait_for_unit("blocky.service")
+        biggerConf.wait_for_open_port(53)
+        biggerConf.wait_for_open_port(5000)
+        biggerConf.succeed("dig @127.0.0.1 printer.lan")
```

Comment:
Maybe something along these lines...
```
dig @127.0.0.1 +short -x 192.168.178.3 | grep -qF printer.lan
```

Reply:
This doesn't actually prove that address mapping works. You should put in what I had.